Geometry/TrackerGeometryBuilder/src/TrackerGeometry.cc

```diff
@@ -129,6 +129,8 @@ void TrackerGeometry::addDetUnitId(DetId p){
 }
 
 void TrackerGeometry::addDet(GeomDet const * p) {
+  // set index
+  const_cast<GeomDet *>(p)->setGdetIndex(theDets.size());
```

Comment:
it would require modifications in the whole TrackerGeometry building code...
let's wait it stabilize for PhaseII, then we can clean it up.